halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/security/X509.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Target, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.model.v1.security;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class X509 extends AuthnMethod {
+
+  private final Method method = Method.X509;
+  private final String nodeName = "x509";
+
+  private String roleOid;
```

Comment:
I'm surprised there's no need for certs in the config?

Reply:
It uses the cert configs from the api ssl settings. We might want to add a validator step to make sure api ssl is turned on then.

Reply:
Ah that's neat. Yeah probably a good idea...

Reply:
I'm going to break that into a separate issue, and go ahead and merge as it is here.